### PR TITLE
Set VirtualGuard do not kill fear if has VectorAPIMethod

### PR DESCRIPTION
**Pull request by r30shah**
If a method we are compining contains the vector API intrinsic method call, set virtual guards kills fear fear flag to false which will prevent merging of Virtual guards with OSR guard and allows expansion of Vector API intrinsic methods.

Signed-off-by: Rahil Shah <rahil@ca.ibm.com>

### CODE REVIEW

**Comment by r30shah** [comment]
@jdmpapin  Can I please get your review on this change? 
FYI @gita-omr @vijaysun-omr

File: runtime/compiler/optimizer/FearPointAnalysis.cpp [hunk 1]
```diff
@@ -34,10 +34,10 @@
 #include "infra/BitVector.hpp"
 #include "infra/Checklist.hpp"
 
-bool TR_FearPointAnalysis::virtualGuardsKillFear()
+bool TR_FearPointAnalysis::virtualGuardsKillFear(TR::Compilation *comp)
    {
    static bool kill = (feGetEnv("TR_FPAnalaysisGuardsDoNotKillFear") == NULL);
-   return kill;
+   return kill && !comp->getMethodSymbol()->hasVectorAPI();
```
Comment (jdmpapin) [inline runtime/compiler/optimizer/FearPointAnalysis.cpp:40]: We should probably check for the `disableVectorAPIExpansion` option here too, like we do for `IfVectorAPI`

**Comment by vijaysun-omr** [comment]
Just checking, did this actually solve the Vector API expansion scenario that led to the discussion that we had on this topic last week ?

**Comment by r30shah** [comment]
> Just checking, did this actually solve the Vector API expansion scenario that led to the discussion that we had on this topic last week ?

Hi @vijaysun-omr , this is the first solution we discussed about last week, It sets do not kill fear for all guards in the method if it contains vector api intrinsic calls. I am working on more granular approach right now, where we do the analysis for every guard and set do not kill fear for virtual guards which guards vector api method only and see if that helps us expanding intrinsic calls or not. 

With the current change, I can get the vectorization to happen without setting `disableOSRGuardMerging` in BumbleBench (Actually with this change, on Z with bumblebench, I do not need to set any JIT option to enforce vector api expansion)

**Comment by jdmpapin** [comment]
Jenkins test sanity all jdk17,jdk19

**Comment by vijaysun-omr** [comment]
Thanks, yes I saw the code was for the "simpler" solution that we had discussed. I was just making sure that it did work. Thanks.

**Comment by jdmpapin** [comment]
AIX JDK17 build failed with "git: not found"

Jenkins test sanity aix jdk17

**Review by jdmpapin — APPROVED** [review]

**Comment by jdmpapin** [comment]
x86-64 JDK19 sanity.functional testList_0 timed out here:
```
17:59:31  ===============================================
17:59:31  Running test JCL_TEST_Java-Security_0 ...
17:59:31  ===============================================
...
18:00:04  ===============================================
18:00:04  GeneralTest suite
18:00:04  Total tests run: 12, Failures: 0, Skips: 0
18:00:04  ===============================================
18:00:04  
02:39:55  Cancelling nested steps due to timeout
```
The previous testList_0 (39) seems not to have run JCL_TEST_Java-Security_0, but 38 did run it, and after the "GeneralTest suite" summary, apparently we should expect to see "JCL_TEST_Java-Security_0_PASSED":
```
[2022-10-20T21:32:01.646Z] ===============================================
[2022-10-20T21:32:01.646Z] Running test JCL_TEST_Java-Security_0 ...
[2022-10-20T21:32:01.646Z] ===============================================
...
[2022-10-20T21:32:31.557Z] ===============================================
[2022-10-20T21:32:31.557Z] GeneralTest suite
[2022-10-20T21:32:31.557Z] Total tests run: 12, Failures: 0, Skips: 0
[2022-10-20T21:32:31.557Z] ===============================================
[2022-10-20T21:32:31.557Z] 
[2022-10-20T21:32:34.132Z] 
[2022-10-20T21:32:34.132Z] JCL_TEST_Java-Security_0_PASSED
```
It's clear that this PR's logic change only affects compilations involving recognized vector API intrinsics. Since the vector API is still in incubation, it really shouldn't have any uses outside of tests that purposefully exercise it, so I don't believe it's possible for this PR to have caused this hang. Restarting.

----

z JDK17 failed JCL_Test_none_SCC_1 with:
```
FAILED: test_interrupted
java.lang.AssertionError: Interrupted returned true for non-interrupted thread
```
This is #16174. Restarting.

----

Jenkins test sanity xlinux,zlinux jdk19

**Comment by jdmpapin** [comment]
https://github.com/eclipse-openj9/openj9/issues/16174 happened again on z, but otherwise everything has passed. Merging.

**Pull request merged**